File: Sorting Plateau/Sorting/Q2. Reduction Operations to Make the Array Elements Equal/Q2_ReductionOperationstoMaketheArrayElementsEqual.py
class Solution(object):
    # Define Solution class

    def reductionOperations(self, nums):
        # Function to count total reduction operations

        nums.sort()
        # Sort the array in ascending order

        size = len(nums)
        # Store total number of elements

        ans = 0
        # Variable to store total operations

        for i in range(size - 1, 0, -1):
            # Loop from last index to first

            if nums[i - 1] != nums[i]:
                # Check if current and previous values are different

                ans += size - i
                # Add count of smaller elements to operations

        return ans
        # Return total operations

File: Sorting Plateau/Sorting/Q2. Reduction Operations to Make the Array Elements Equal/Q2_ReductionOperationstoMaketheArrayElementsEqual.java
class Solution {
    // Solution class

    public int reductionOperations(int[] nums) {
        // Method to count total reduction operations

        Arrays.sort(nums);
        // Sort the array in ascending order

        int si = nums.length;
        // Store total number of elements

        int ans = 0;
        // Stores total operations count

        for (int i = nums.length - 1; i > 0; i--) {
            // Loop from largest element to smallest

            if (nums[i - 1] != nums[i]) {
                // Check if current and previous elements are different

                ans += si - i;
                // Add number of smaller elements to operations
            }
        }

        return ans;
        // Return total operations
    }
}